### ISSUE TITLE
Can't download add-on because of wrong hash.

### ISSUE THREAD

**Issue by Myer** [Quicksaver/FindBar-Tweak#245]
Version: Firefox 44.0.2
Build-ID: 20160210153822
User-Agent: Mozilla/5.0 (Windows NT 6.3; WOW64; rv:44.0) Gecko/20100101 Firefox/44.0

When I try to download the new version 2.1.9 through "addons.mozilla.org" I get the following message:
"The add-on could not be installed because it does not match the add-on Firefox expected."

Th error-console shows me the following:
1457030192297	addons.xpi	WARN	Download of https://addons.mozilla.org/firefox/downloads/file/404939/findbar_tweak-2.1.9-fx.xpi?src=version-history failed: Downloaded file hash (465c345024d68bb670c6698b202e4f2dc1ead665e404f5ad3adf99595d6d58fe) did not match provided hash (2eaf4fd6071dea122eaf977e01f9cb7b0f3f22655b822cb7f80bb08691ef88e7)

Could it be that you uploaded the wrong file?

Btw: Updating it via "about:addons" doesn't work either.
Also: I don't think this has anything to do with add-on signing but just-in-case, the setting "xpinstall.signatures.required" is set to "false".

**Comment by Myer** [comment]
I've manually downloaded the xpi and yes, the SHA-256 is really:
"2eaf4fd6071dea122eaf977e01f9cb7b0f3f22655b822cb7f80bb08691ef88e7"

So I guess somewhere on "addons.mozilla.org" you have provided them the wrong hash. Not sure how the update-process actually works...

**Comment by Quicksaver** [comment]
This is apparently a known problem that AMO has been having lately. They're still not sure what is causing it though. I was hoping you could help provide a little more information if you can. Could you try installing the latest version directly from the listing page at AMO: https://addons.mozilla.org/firefox/addon/findbar-tweak/

In case it fails again, please include the response headers. You can do so by:
* keep that add-on listing page open
* open the Firefox developer tools (Ctrl+Shift+K), go to Settings pane (usually the fourth icon from the end)
  * check "enable browser chrome and add-on debugging toolboxes" and "Enable remote debugging"
* close developers tools pane
* go to Tools -> Web developer -> Browser toolbox (or Ctrl+Alt+Shift+I)
* confirm the warning by clicking ok
* go to the network pane
* back on the add-on page, click "Add to Firefox"
* back in the network pane of the toolbox, you should see at least one request appearing
* select the one that is not a redirect (the one for addons.cdn.mozilla.net)
* on the right hand side, you see the headers, click "Raw headers"
* copy the "response headers"

Copy the results here, or better yet if you can, paste them directly in https://github.com/mozilla/addons-server/issues/1111

**Comment by Myer** [comment]
Well... I'm not sure why but now it works. :-/

I was enabling the remore debugging and clicked on "Add to Firefox" and then the request popped up but it also was succesfull. Maybe I shouldn't have reloaded the add-on page. Guess I was using another download-server or something.

**Comment by Quicksaver** [comment]
That's definitely weird, reloading the page shouldn't have mattered, in theory at least, unless there were some odd cache issues.

Well at least you managed to update it. :) I'll leave this open in case others run into it (you're not the first one to report this in the last few days), to see if we can get more info.

**Comment by Myer** [comment]
Btw: I'm using very long Firefox sessions together with many many tabs. If I remember correctly I have already tried to update this add-on via "about:addons" some days ago. So I guess all other requests for the newer version were actually using the cashed version.

Only now I went to the add-on page on "addons.mozilla.org" and tried downloading it from there.
And before I tried your suggestion using remote debugging I thought "maybe first clearing cache" so I reloaded the page with [CTRL]+[F5].
So I guess only now after clicking the "Add to Firefox" button I actually got the non-corrupted xpi since there was no cached version available.

But that's just some idea...
